src/mame/philips/patgen.cpp

```diff
@@ -0,0 +1,422 @@
+ROM_END
+
+ROM_START( pm5644p00 )
+	ROM_REGION( 0x10000, "maincpu", 0 )
+	ROM_LOAD( "4008_102_59391_csum_0d00.v12",  0x000000, 0x10000, CRC(9e30f0dd) SHA1(3277db6f5c890897c6454b41a60103de6f652690) )
+
+	ROM_REGION( 0x50000, "luma", 0 )
+	ROM_LOAD( "4008_102_56831_csum_6c99.v101",  0x000000, 0x10000, CRC(dce2ec5a) SHA1(5fd53a091b39164a36c3109bac41d7f80f50c982) )
+	ROM_LOAD( "4008_102_56841_csum_6cf9.v103",  0x010000, 0x10000, CRC(e31a3ebc) SHA1(24e7580ebf9d10f21a3a207b73b404f1c07300e0) )
+	ROM_LOAD( "4008_102_56851_csum_0379.v105",  0x020000, 0x10000, CRC(73ef8961) SHA1(03f8703df4f1c13a70b0c397a26eff8308a27685) )
+	ROM_LOAD( "4008_102_56861_csum_0393.v107",  0x030000, 0x10000, CRC(fe4477d4) SHA1(998c4a28c3a80e756cdaa1016b0019bc47e9995f) )
+	ROM_LOAD( "4008_102_57161_csum_e601.v109",  0x040000, 0x10000, CRC(3e71ec87) SHA1(8342e0bbc768dd70ce5d484dd3e0f39a83c7b9a5) )
+
+	ROM_REGION( 0x20000, "chromav", 0 )
+	ROM_LOAD( "4008_102_57171_csum_d3b6.v201",  0x000000, 0x10000, CRC(6bab160c) SHA1(297f35e39adc6794bda82ca351cb9c7f8499f1f3) )
+	ROM_LOAD( "4008_102_57181_csum_de42.v203",  0x010000, 0x10000, CRC(d849cbf4) SHA1(586251fc76f6ec5fa3bca87a3115d594536d41e3) )
+
+	ROM_REGION( 0x20000, "chromau", 0 )
+	ROM_LOAD( "4008_102_57191_csum_f178.v301",  0x000000, 0x10000, CRC(378419c7) SHA1(6a1aaf6c20e9ef13a1a72a787b1822a4118d3f68) )
+	ROM_LOAD( "4008_102_57201_csum_e138.v303",  0x010000, 0x10000, CRC(80c9b2fe) SHA1(c0241458a51a9790ed4c977f667a94a40e612982) )
+
+ROM_END
+
+} // anonymous namespace
+
+SYST( 1989, pm5644g00,   0,           0, patgen, patgen, patgen_state, empty_init, "Philips", "PM 5644 color pattern generator (PAL-G 4:3 Colour Circle)", MACHINE_NO_SOUND | MACHINE_NOT_WORKING)
+SYST( 1989, pm5644g913,  0,           0, patgen, patgen, patgen_state, empty_init, "Philips", "PM 5644 color pattern generator (PAL-G 4:3 Indian Head)", MACHINE_NO_SOUND | MACHINE_NOT_WORKING)
+SYST( 1989, pm5644g924,  0,           0, patgen, patgen, patgen_state, empty_init, "Philips", "PM 5644 color pattern generator (PAL 16:9 Colour Circle Rev 1)", MACHINE_NO_SOUND | MACHINE_NOT_WORKING)
+SYST( 1989, pm5644g924b, pm5644g924,  0, patgen, patgen, patgen_state, empty_init, "Philips", "PM 5644 color pattern generator (PAL 16:9 Colour Circle Rev 2)", MACHINE_NO_SOUND | MACHINE_NOT_WORKING)
+SYST( 1989, pm5644l00,   0,           0, patgen, patgen, patgen_state, empty_init, "Philips", "PM 5644 color pattern generator (SECAM 4:3 Colour Circle)", MACHINE_NO_SOUND | MACHINE_NOT_WORKING)
+SYST( 1989, pm5644m00,   0,           0, patgen, patgen, patgen_state, empty_init, "Philips", "PM 5644 color pattern generator (NTSC 4:3 Colour Circle)", MACHINE_NO_SOUND | MACHINE_NOT_WORKING)
+SYST( 1989, pm5644p00,   0,           0, patgen, patgen, patgen_state, empty_init, "Philips", "PM 5644 color pattern generator (PAL-M 4:3 Colour Circle)", MACHINE_NO_SOUND | MACHINE_NOT_WORKING)
```

Comment:
Does this system even have any sound capability? If no then it should be `MACHINE_NO_SOUND_HW` in place of `MACHINE_NO_SOUND`